src/include/m/utility/pointers.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <exception>
+#include <stdexcept>
+#include <type_traits>
+
+namespace m
+{
+    template <typename T>
+        requires std::is_pointer_v<T>
+    struct not_null
+    {
+        not_null() = delete;
+
+        not_null(T v): m_v(v)
+        {
+            if (v == nullptr)
```

Comment:
Is this worth some sort of one-liner? A macro?
Delete the newline? clang-format disallows?